src/types.cpp

```diff
@@ -13,11 +13,11 @@ Input &Input::operator>>(string &val)
 Input Input::fromHere()
 {
     string cmd;
-    if (!(*this >> cmd)) {
+    if (empty()) {
         return {{}, {}};
     }
 
-    return Input(cmd, toVector());
+    return Input(*(begin()), Container(begin() + 1, end()));
```

Comment:
It is not really clear to me how does this change corresponds to this PR?

Reply:
This just fixes the behaviour of `Input::fromHere()` such that it can be called multiple times. Before, calling `fromHere()` consumed the command name but not the args, now it does not consume anything. We didn't notice this issue before, because before this PR, we never used the Input after calling `fromHere()`.

Reply:
I see, thanks!